HW-8 Koin 적용

BeokBeok opened the pull request:
Koin 적용 하였습니다.

BeokBeok committed:
HW-8 Koin dependency 추가

BeokBeok committed:
HW-8 Application 클래스 추가 및 Koin 등록

BeokBeok committed:
HW-8 Koin 을 이용하여 DI 적용

BeokBeok committed:
HW-8 패키지 정리

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/di/UpbitModules.kt:
```diff
@@ -0,0 +1,68 @@
+    viewModel { UpbitViewModel(get()) }
+}
+
+val dataSourceModules = module {
```
분리시키는 게 어떨까요?

BeokBeok replied:
Upbit 관련 모듈들을 한 곳에 모아서 처리하였는데, 어떻게 분리시켜야할지 조언 부탁드립니다.

NackUn replied:
@BeokBeok upbit 관련 모듈들 중 LocalServiceModules, RemoteServiceModules 는 분리시켜서 재사용할 수 있게 만들면 어떨까요..?

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/data/upbit/source/UpbitRepository.kt:
```diff
@@ -5,7 +5,7 @@ import com.architecturestudy.data.upbit.source.local.UpbitLocalDataSource
 import io.reactivex.disposables.Disposable
 
-class UpbitRepository private constructor(
+class UpbitRepository constructor(
```
constructor를 써준 이유가 뭔가요??

BeokBeok replied:
8095b30063620f0e196b9a2b4c2dc42fcd6afde6
제거 하였습니다.

NackUn commented on BeokBeok/app/src/main/java/com/architecturestudy/base/BaseFragment.kt:
```diff
@@ -10,12 +10,13 @@ import androidx.databinding.DataBindingUtil
 import androidx.fragment.app.Fragment
 
-abstract class BaseFragment<B : ViewDataBinding>(
+abstract class BaseFragment<VDB : ViewDataBinding, VM : BaseViewModel>(
```
ViewModel 을 공유하려고 추가해주신건가요?

BeokBeok replied:
Fragment 에서 ViewModel 을 생성하고, 각 Fragment 들은 BaseFragment 를 상속받기 때문에 추가하였습니다.

BeokBeok committed:
HW-8 constructor 제거

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/146#discussion_r309051866

sjjeong commented on BeokBeok/app/src/main/java/com/architecturestudy/di/UpbitModules.kt:
```diff
@@ -0,0 +1,68 @@
+
+}
+
+fun getLocalServiceModules(context: Context) = module {
```
context를 받아서 넣는 방법 말고 applicationContext를 가져오는 방법이 있습니다.

BeokBeok replied:
this@Application 을 전달하면 application context로 전달 되는 것으로 알고 있었습니다.

수정 하였습니다.
3117f10632a66051a93716b4f1147331fbd78bf4

sjjeong replied:
androidApplication()를 사용하시면 됩니다.

BeokBeok replied:
적용하였습니다.
e5dcccfd4daf4e0d5b3b2642bd4328e81a195681

BeokBeok committed:
HW-8 application context 를 적용하도록 변

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/146#discussion_r309723592

BeokBeok committed:
HW-8 context 대신 androidApplication 을 사용하도록 변경

- Review URL
https://github.com/StudyFork/GoogryAndroidArchitectureStudy/pull/146#discussion_r310336269

sjjeong reviewed (approved):
LGTM ✊ ✌️ 👋
